### PR TITLE
Created endpoints module

### PR DESCRIPTION
**Pull request by Chimexx**
-Added endpoints module
-Removed error object from try-catch
-Changed some columns in endpoints entity to enum
-Removed endpointsId from Api entity

### CODE REVIEW

File: src/endpoints/endpoints.service.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      const newEndpoint = this.endpointRepository.create({
+        ...createEndpointDto,
+        apiId,
+      });
+
+      const savedEndpoint = await this.endpointRepository.save(newEndpoint);
+      return savedEndpoint;
+    } catch (error) {
+      throw new BadRequestException(
+        ZapiResponse.BadRequest('Server error', '500 Internal Server Error'),
```
Comment (DorcasOJ) [inline src/endpoints/endpoints.service.ts:47]: ZapiResponse.BadRequest('Server error', error, '500'),

Comment (Chimexx) [reply src/endpoints/endpoints.service.ts:47]: It was so Initially, but brain doesn't want the whole error object showing up on the frontend...

Comment (PhilemonBrain) [reply src/endpoints/endpoints.service.ts:47]: u can look at the error response and return only the message or a non sensitive info. All that info isn't really necessary

File: src/entities/api.entity.ts [hunk 2]
```diff
@@ -77,10 +73,6 @@ export class Api extends SharedEntity {
   @JoinColumn({ name: 'tutorialsId' })
```
Comment (PhilemonBrain) [inline src/entities/api.entity.ts:73]: is it possible to have this JoinColumn, without a relationship

**Review by PhilemonBrain — CHANGES_REQUESTED** [review]

**Comment by Chimexx** [comment]
- Fixed error response
- removed relationship between tutorials and Api
- Updated migrations

**Comment by Chimexx** [comment]
@PhilemonBrain  is the update okay?

File: src/entities/api.entity.ts [hunk 3]
```diff
@@ -73,14 +68,6 @@ export class Api extends SharedEntity {
   @JoinColumn({ name: 'discussionsId' })
```
Comment (PhilemonBrain) [inline src/entities/api.entity.ts:68]: please look at this. 
https://orkhan.gitbook.io/typeorm/docs/relations#joincolumn-options
i dont think u can use JoionColumn without a relationoship

File: src/api/api.service.ts [hunk 4]
```diff
@@ -38,7 +38,13 @@ export class ApiService {
       return await this.apiRepository.save(newApi);
     } catch (error) {
       throw new BadRequestException(
-        ZapiResponse.BadRequest('Server error', error, '500'),
+        ZapiResponse.BadRequest(
+          'Internal Server error',
+          error instanceof Error
```
Comment (PhilemonBrain) [inline src/api/api.service.ts:43]: i think u can just say error.message for now

Comment (PhilemonBrain) [reply src/api/api.service.ts:43]: and also it's replicated in numerous places, ignore that for now. But in the production level app, we wont leave it like this

**Review by PhilemonBrain — CHANGES_REQUESTED** [review]

**Pull request merged**